### PR TITLE
[iOS][Android] DatePicker: Added a new UseNativeMinMaxDates property

### PR DESCRIPTION
**Pull request by carldebilly**
Fixes https://github.com/unoplatform/uno/issues/5575

# Feature

Added a new UseNativeMinMaxDates property on DatePicker & NativeDatePickerFlyout to allow the picker to be restricted to some date instead of just the year.  Properties `MinYear` and `MaxYear` are still used for that since they are already of type `DateTimeOffset`.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [X] Contains **NO** breaking changes
- [X] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [X] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/DatePicker/NativeDatePickerFlyout.iOS.cs [hunk 1]
```diff
@@ -36,6 +36,28 @@ private NativeDatePickerFlyoutPresenter _presenter
...
+		/// </summary>
+		public bool UseNativeMinMaxDates
+		{
+			get => (bool)GetValue(UseNativeMinMaxDatesProperty);
+			set => SetValue(UseNativeMinMaxDatesProperty, value);
+		}
+
+		private static void OnUseNativeMinMaxDatesChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
+		{
+			if (o is NativeDatePickerFlyout selector)
```
[the remark this answers is not in the document]
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/DatePicker/NativeDatePickerFlyout.iOS.cs:56]: Oohh nice catch (not you, Codacy...)

[the remark this answers is not in the document]
Comment (carldebilly) [reply src/Uno.UI/UI/Xaml/Controls/DatePicker/NativeDatePickerFlyout.iOS.cs:56]: Somethig strange...  it's defenitely not empty. Weird it is here.

**Comment by jeromelaban** [comment]
@carldebilly Documentation is missing to explain how the flag works

**Comment by YGuerin** [comment]
@jeromelaban @carldebilly 
Was this backported to Uno.UI 3.8? If not, can it be?
Thank you

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/3.8

**Pull request merged**